src/main/kotlin/com/zepben/evolve/database/sqlite/metrics/MetricsDataSourceWriter.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2024 Zeppelin Bend Pty Ltd
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package com.zepben.evolve.database.sqlite.metrics
+
+import com.zepben.evolve.database.sqlite.common.SchemaUtils
+import com.zepben.evolve.database.sqlite.common.TableVersion
+import com.zepben.evolve.database.sqlite.extensions.configureBatch
+import com.zepben.evolve.metrics.IngestionJob
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.sql.Connection
+import javax.sql.DataSource
+
+/**
+ * Class for writing metrics to an arbitrary datasource.
+ */
+class MetricsDataSourceWriter @JvmOverloads constructor(
```

Comment:
Discussion: It's a shame this doesn't inherit from `BaseDatabaseWriter`, but unfortunately it has sqlite built in. That said I feel it's also a shame this doesn't mimic the public interface of that class (although I prefer the save taking the data to be saved like you've done here rather than the constructor on the sqlite ones...) 
All that said, if we don't mind breaking changes, you could actually make this and the sqlite writers inherit from the same base class I reckon. However, I probably care more about consistent public API then most...

Reply:
Yep, this is all public API, not private code, so we should be aiming for consistency.

I would prefer a breaking change to introduce a new layer for SQLite base writer to allow this to inherit from the same parent class if that is required.